app/Repositories/ProfileRepository.php

```diff
@@ -80,19 +80,12 @@ public function getProfilesByGroup($site_id)
         $all_profiles = $this->getProfiles($site_id, $group_ids);
 
         // Organize profiles by the group they are in keyed by accessid
-        $grouped = collect($all_profiles['profiles'])->map(function ($profile) {
-            return collect($profile['groups'])->flatMap(function ($group) use ($profile) {
-                return [
-                    'data' => $profile['data'],
-                    'groups' => $profile['groups'],
-                    'group' => $group,
-                    'AccessID' => $profile['data']['AccessID'],
-                    'link' => $profile['link'],
-                ];
-            });
-        })
-        ->keyBy('AccessID')
-        ->groupBy('group', true)
+        $grouped = collect($all_profiles['profiles'])->keyBy('data.AccessID')
+        ->groupBy([
+            function ($profile) {
+                return $profile['groups'];
+            },
+        ], $preserveKeys = true)
```

Comment:
We talked about changing this away from the assignment format and instead pass the boolean:

`], true) // Preserve Keys`